test/common/stats/thread_local_store_test.cc

```diff
@@ -1149,6 +1149,36 @@ TEST_F(StatsMatcherTLSTest, RejectPrefixNoDot) {
   EXPECT_MEMORY_LE(mem_consumed, 3500000);
 }
 
+TEST_F(StatsMatcherTLSTest, DoNotRejectHiddenRegex) {
+  InSequence s;
+
+  stats_config_.mutable_stats_matcher()->mutable_exclusion_list()->add_patterns()->MergeFrom(
+      TestUtility::createRegexMatcher(".*[A-Z].*"));
+  store_->setStatsMatcher(std::make_unique<StatsMatcherImpl>(stats_config_, symbol_table_));
+
+  Gauge& lowercase_hidden_gauge =
+      scope_.gaugeFromString("lowercase_hidden_gauge", Gauge::ImportMode::Hidden);
+  EXPECT_EQ(lowercase_hidden_gauge.name(), "lowercase_hidden_gauge");
+  Gauge& uppercase_hidden_gauge =
+      scope_.gaugeFromString("lowercase_HIDDEN_gauge", Gauge::ImportMode::Hidden);
+  EXPECT_EQ(uppercase_hidden_gauge.name(), "lowercase_HIDDEN_gauge");
```

Comment:
I think you can replace one of these with a case where the ImportMode is something other than hidden, and have the test validate that hte returned gauge is the null gauge.

Reply:
Done